### PR TITLE
[WIP] Add error tests

### LINKED ISSUE #67: Generify errno tests
Most errno tests are pratically the same, therefore a set of assertion helpers would drastically reduce the amount of code to rewrite.
The most obvious example is ELOOP.

### PR DESCRIPTION
**Pull request by musikid**
Close #67

### CODE REVIEW

File: rust/src/main.rs [hunk 1]
```diff
@@ -63,13 +63,13 @@ fn main() -> anyhow::Result<()> {
...
     }
 
-    let config: Config = if let Some(path) = args.configuration_file.as_deref() {
-        Figment::new().merge(Toml::file(path))
-    } else {
-        Figment::new()
-    }
-    .extract()
-    .unwrap_or_default();
+    let config: Config = Figment::new()
```
Comment (asomers) [inline rust/src/main.rs:66]: Why this change?  This will cause an error if no config file is specified and $PWD/pjdfstest.toml does not exist.

File: rust/src/tests/errors.rs [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    path::{Path, PathBuf},
+};
+
+use crate::{
+    runner::context::{FileType, SerializedTestContext, TestContext},
+    tests::{chmod, lchmod, lchown, link, rename, symlink},
+};
+
+crate::test_case! {eloop}
+fn eloop(ctx: &mut TestContext) {
```
Comment (asomers) [inline rust/src/tests/errors.rs:31]: This test combines way too many assertions into a single test case.  Could you split it up?

File: rust/src/tests/errors.rs [hunk 3]
```diff
@@ -0,0 +1,589 @@
...
+    fn assert_enotdir<T: Debug, F>(path: &Path, f: F)
+    where
+        F: Fn(&Path) -> nix::Result<T>,
+    {
+        assert_eq!(f(path).unwrap_err(), Errno::ENOTDIR);
+    }
+
+    // TODO: Add rmdir upstream
+    // assert_enotdir(&path, |p| rmdir(p, Mode::empty()));
+    assert_enotdir(&path, |p| chflags(p, FileFlag::empty()));
```
Comment (asomers) [inline rust/src/tests/errors.rs:116]: The assert_enotdir function doesn't buy much.  I think it's more readable just to inline it:
```rust
assert_eq!(chmod(p, Mode::empty()), Error::ENOTDIR);
```

**Review by asomers — CHANGES_REQUESTED** [review]

File: rust/src/main.rs (hunk 1 above)
Comment (musikid) [reply rust/src/main.rs]: Seems to be a merge error.

**Comment by musikid** [comment]
I will reopen it when it will be in a better shape, because it's still messy for now...
